### DISCUSSION TITLE
Bundle TS code without compiling it to JS

### DISCUSSION THREAD

**Discussion by vyshkov** [webpack/webpack#19002] [General]
I am using Webpack to bundle node.js app so I won't need to deploy node_modules to my serverless infrastructure.

Since Node.js 22 (LTS) [officially supports Typescript](https://nodejs.org/docs/latest-v22.x/api/typescript.html#full-typescript-support), it now makes sense to only bundle node_modules dependencies without compiling everything to JS.

Theoretically, it should reduce build time.

Is it possible to achieve this with Webpack? Do you think it will have any effect on the app build time and execution performance?

**Comment by absidue** [comment]
The conversion of TypeScript to JavaScript still happens, it's just that instead of doing the type checking and conversion to JavaScript at build time, you are moving it to runtime (Node.js will have to do the extra step of stripping the types before it can run the code). So faster builds but slower app startup.

**Comment by vyshkov** [comment]
So it doesn't make sense for a serverless infra because it may increase cold start time.
